File: app/frontend/shared/application/trigger-finder.ts
import { TriggerRepository } from '../domain/trigger-repository'
import { Trigger, TriggerId } from '../domain/trigger'

export class TriggerFinder {

  constructor(private repository: TriggerRepository) { }

  async findByDiscordServerId(discordServerId: string): Promise<Trigger[]> {
    return await this.repository.findByDiscordServerId(discordServerId)
  }

  async findById(uuid: TriggerId): Promise<Trigger> {
    return await this.repository.findById(uuid)
  }
}

File: context/src/main/kotlin/com/usadapekora/context/application/trigger/find/TriggerFinder.kt
package com.usadapekora.context.application.trigger.find

import com.usadapekora.context.domain.trigger.Trigger
import com.usadapekora.context.domain.trigger.TriggerException
import com.usadapekora.context.domain.trigger.TriggerMatcher
import com.usadapekora.context.domain.trigger.TriggerRepository

class TriggerFinder(
    private val repository: TriggerRepository,
    private val matcher: TriggerMatcher
) {
    fun findByInput(input: String, discordServerId: String): TriggerResponse {
        val triggers = repository.findByDiscordServer(Trigger.TriggerDiscordGuildId(discordServerId))
        return TriggerResponse.fromEntity(matcher.matchInput(input, triggers) ?: throw TriggerException.NotFound())
    }

    fun find(id: String): TriggerResponse
        = TriggerResponse.fromEntity(repository.find(Trigger.TriggerId(id)))

    fun findByDiscordServer(discordServerId: String): TriggersResponse
        = TriggersResponse.fromArray(repository.findByDiscordServer(Trigger.TriggerDiscordGuildId(discordServerId)))
}
